sonic-pcied/scripts/pcied

```diff
@@ -157,6 +160,31 @@ class DaemonPcied(daemon_base.DaemonBase):
         if self.resultInfo is None:
             return
 
+        try:
+            stream = open(PLATFORM_PCIE_YAML_FILE, 'r')
+            dictionary = yaml.safe_load(stream)
+            for elem in dictionary:
+                bus = int(elem.get('bus'), 16)
+                device = int(elem.get('dev'), 16)
+                func = int(elem.get('fn'), 16)
+                pcieDeviceID = elem.get('id')
+
+                pcie_device = ("{:02x}:{:02x}.{:01d}".format(bus, device, func))
+                cmd = ["setpci", "-s", pcie_device, "2.w"]
+                output = subprocess.check_output(cmd)
+                pcieDeviceQueryResult = output.decode('utf8').rstrip()
+
+                if pcieDeviceQueryResult is None or "No devices selected" in pcieDeviceQueryResult:
+                    self.log_error("PCIe Peripheral {} -- Not Found".format(pcieDeviceQueryResult))
+                elif pcieDeviceQueryResult != pcieDeviceID:
+                    if pcieDeviceQueryResult == 'ffff':
```

Comment:
what if the device id != 0xffff and still does not match the pice.yaml file content?

Reply:
Good point - will modify the logic to handle this use case

Reply:
**elif pcieDeviceQueryResult != pcieDeviceID:**
                    if pcieDeviceQueryResult == 'ffff':
                        self.log_error("PCIe device {:02x}:{:02x}.{:01d} missing.".format(bus, device, func))
                    **else:
                        self.log_error("PCIe device {:02x}:{:02x}.{:01d} ID mismatch. Expected {}, received {}".format(bus, device, func, pcieDeviceID, pcieDeviceQueryResult))**
                        
This scenario is handled by the bold-ed code -- if queried value isn't expected value it enters the if block. If it is 'ffff' it declares the device as missing. However, if the queried value is not ffff, that means it is some other value that does not match the yaml file content, so it falls to the 'else' and declares a mismatch.

Reply:
Modified deviceID of a device in pcie.yaml file to test this - here is the log:

Mar 15 20:48:32.836090 str-s6000-on-6 INFO pmon#supervisord 2023-03-15 20:48:32,834 INFO success: pcied entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)

Mar 15 20:49:24.162340 str-s6000-on-6 ERR pmon#pcied[66]: PCIe device 00:02.0 ID mismatch. Expected 0b47, received 0c47